app/models/manageiq/providers/hawkular/middleware_manager/refresh_parser.rb

```diff
@@ -34,7 +34,13 @@ def fetch_middleware_servers
             host_instance = find_host_by_bios_uuid(machine_id) ||
                             find_host_by_bios_uuid(alternate_machine_id(machine_id))
 
-            if host_instance
+            if server[:properties].try { |prop| prop['In Container'] } == 'true'
```

Comment:
Looks like a good place to use `fetch_path`, try `server.fetch_path(:properties, 'In Container') == true`
Using a method could help readability as well